From c62675e1bc14b35bb43134f98daa70d13bd6e7b0 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Wed, 13 Feb 2019 11:29:34 +0200
Subject: [PATCH 1001/1051] arm64: dts: marvell: 91xx: add aliases to network
 nodes

This patch fixes random MAC addresses on network interfaces.
Network nodes in Linux device tree are mapped to ethernetX by
aliases.
EthernetX represent ethXaddr environment associated with
probed network interfaces in bootloader.
MAC address is picked up from device tree during port probe
procedure.

Change-Id: Ie500c82265415ad0a98841bce753790c1496438a
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/3933
Reviewed-by: Grzegorz Jaszczyk <Grzegorz.Jaszczyk@cavium.com>
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
Tested-by: sa_ip-sw-jenkins
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/cn9130-db-A.dts | 3 +++
 arch/arm64/boot/dts/marvell/cn9131-db.dtsi  | 2 ++
 arch/arm64/boot/dts/marvell/cn9132-db-A.dts | 1 +
 3 files changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/cn9130-db-A.dts b/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
index d4652c6c012b..b2b91da545f8 100644
--- a/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/cn9130-db-A.dts
@@ -18,6 +18,9 @@
 
 	aliases {
 		i2c0 = &cp0_i2c0;
+		ethernet0 = &cp0_eth0;
+		ethernet1 = &cp0_eth1;
+		ethernet2 = &cp0_eth2;
 	};
 
 	memory@00000000 {
diff --git a/arch/arm64/boot/dts/marvell/cn9131-db.dtsi b/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
index 8de19096525f..7c3c48df6b72 100644
--- a/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/cn9131-db.dtsi
@@ -28,6 +28,8 @@
 	aliases {
 		gpio3 = &cp1_gpio1;
 		gpio4 = &cp1_gpio2;
+		ethernet3 = &cp1_eth0;
+		ethernet4 = &cp1_eth1;
 	};
 
 	cp1_reg_usb3_vbus0: cp1_usb3_vbus@0 {
diff --git a/arch/arm64/boot/dts/marvell/cn9132-db-A.dts b/arch/arm64/boot/dts/marvell/cn9132-db-A.dts
index b949590bb745..f82194329c20 100644
--- a/arch/arm64/boot/dts/marvell/cn9132-db-A.dts
+++ b/arch/arm64/boot/dts/marvell/cn9132-db-A.dts
@@ -16,6 +16,7 @@
 	aliases {
 		gpio5 = &cp2_gpio1;
 		gpio6 = &cp2_gpio2;
+		ethernet5 = &cp2_eth0;
 	};
 };
 
-- 
2.17.1

